frame/crowdloan-rewards/src/lib.rs

```diff
@@ -0,0 +1,325 @@
+			let total_contributors = rewards.len();
+			let total_rewards = rewards.into_iter().fold(
+				T::Balance::zero(),
+				|total, (remote_account, account_reward, vesting_period)| {
+					Rewards::<T>::insert(
+						remote_account,
+						Reward {
+							total: account_reward,
+							claimed: T::Balance::zero(),
+							vesting_period,
+						},
+					);
+					total + account_reward
+				},
+			);
+			TotalRewards::<T>::set(total_rewards);
+			TotalContributors::<T>::set(total_contributors as u32);
+		}
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		NotInitialized,
+		AlreadyInitialized,
+		InvalidProof,
+		InvalidClaim,
+		NotAContributor,
+		NothingToClaim,
+	}
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		#[allow(missing_docs)]
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+		type Currency: Mutate<Self::AccountId, Balance = Self::Balance>;
+		type Balance: Default
+			+ Parameter
+			+ Codec
+			+ Copy
+			+ Ord
+			+ CheckedAdd
+			+ CheckedSub
+			+ CheckedMul
+			+ AtLeast32BitUnsigned
+			+ MaybeSerializeDeserialize
+			+ Zero;
+		type AdminOrigin: EnsureOrigin<Self::Origin>;
+		type Convert: Convert<Self::BlockNumber, Self::Balance>;
+		type RelayChainAccountId: Parameter + MaybeSerializeDeserialize + Into<AccountId32> + Ord;
+		#[pallet::constant]
+		type InitialPayment: Get<Perbill>;
+		#[pallet::constant]
+		type VestingPartition: Get<Self::BlockNumber>;
+		#[pallet::constant]
+		type Prefix: Get<&'static [u8]>;
+	}
+
+	#[pallet::storage]
+	pub type Rewards<T: Config> =
+		StorageMap<_, Blake2_128Concat, RemoteAccountOf<T>, RewardOf<T>, OptionQuery>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn total_rewards)]
+	pub type TotalRewards<T: Config> = StorageValue<_, T::Balance, ValueQuery>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn claimed_rewards)]
+	pub type ClaimedRewards<T: Config> = StorageValue<_, T::Balance, ValueQuery>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn total_contributors)]
+	pub type TotalContributors<T: Config> = StorageValue<_, u32, ValueQuery>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn vesting_block_start)]
+	pub type VestingBlockStart<T: Config> = StorageValue<_, T::BlockNumber, ValueQuery>;
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	pub struct Pallet<T>(_);
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		#[pallet::weight(10_000)]
+		pub fn initialize(origin: OriginFor<T>) -> DispatchResult {
+			T::AdminOrigin::ensure_origin(origin)?;
+			ensure!(!VestingBlockStart::<T>::exists(), Error::<T>::AlreadyInitialized);
+			let current_block = frame_system::Pallet::<T>::block_number();
+			VestingBlockStart::<T>::set(current_block);
+			Ok(())
+		}
+
+		#[pallet::weight(10_000)]
+		pub fn claim_from_ethereum(
+			origin: OriginFor<T>,
+			reward_account: T::AccountId,
+			ethereum_address: EthereumAddress,
+			proof: EcdsaSignature,
+		) -> DispatchResult {
+			ensure_signed(origin)?;
```

Comment:
why would you want this to be signed?

Reply:
we don't but I have setup the extrinsic as is